docs/ruleset_functions/normalize_space_schedules.md

```diff
@@ -0,0 +1,43 @@
+
+## normalize_space_schedules
+
+Description: This function would determine a normalized schedule for a data element in space.  
+
+Inputs:
+  - **Space.data_element**: The space data element that needs to determine a normalized schedule, e.g. space.interior_lighting.
+
+Returns:
+- **space_normalized_schedule_array**: An array containing 8760 hourly values of a normalized schedule of the space data element.
+
+Function Call:
+
+- convert_schedule_to_hourly()
+
+Applicability:
+
+- space.interior_lighting
+- space.miscellaneous_equipment (To be developed)
+
+Logic:
+
+- For each interior lighting in space: `for interior_lighting in space.interior_lighting:`
+
+  - Get lighting power per area: `power_per_area = interior_lighting.power_per_area`
+
+  - Add lighting power per area to space total: `space_total_power_per_area += power_per_area`
+
+  - Get lighting schedule: `schedule = interior_lighting.lighting_multiplier_schedule`
+
+    - If schedule is hourly type, multiply hourly values with power per area and save to an 8760 hours array: `if schedule.schedule_sequence_type == "HOURLY": 8760_array.append(hourly_value * power_per_area for hourly_value in schedule.hourly_values)`
+
+    - Else, schedule is event type, convert schedule values to hourly: `else: converted_schedule_array = convert_schedule_to_hourly(schedule)`
+
```

Comment:
I don't know what schedule_sequence_type=event means. Since we expect this function to be applicable to lighting and misc. equipment, should only "hourly" be allowed here? @supriyagoel if you agree perhaps this may be another general RCT check.

Reply:
It seems a schedule can be in either hourly or event format. I'm not sure if this event format is applicable to all type of schedules or just some special types.

Reply:
Could you help me understand what event format means? Thanks!

Reply:
It seems an event type schedule has two fields, event_time (when the schedule value changes) and event_values (schedule value). So instead of describing all 8760 hour values, it just reports a few time spot when the value changes (as a simplified way I assume?).

Reply:
My only outstanding comment is that convert_schedule_array function has to be developed to convert event schedule to hourly schedule.